tests/unit/Term/TermFallbackTest.php

```diff
@@ -53,11 +54,20 @@ public function testGivenInvalidSourceLanguageCode_constructorThrowsException( $
 
 	public function invalidSourceLanguageCodeProvider() {
 		return array(
-			array( 21 ),
-			array( '' ),
+			'integer' => array( 21 ),
+			'emptyString' => array( '' ),
+			'sameAsRequestedLanguageCode' => array( 'foor' ),
 		);
 	}
 
+	/**
+	 * @expectedException InvalidArgumentException
+	 */
+	public function testGivenInvalidRequestedSameAsActualAndNoTransliteration_constructorThrowsException() {
```

Comment:
I suggest to write individual tests like this one, either alternatively or in addition to the test cases in the providers. I come up with the following relevant combinations:
* `( 'requested', 'text', 'actual', 'source' )` => valid
* `( 'requested', 'text', 'actual', null )` => valid
* `( 'requested', 'text', 'actual', 'actual' )` => exception
* `( 'requested', 'text', 'actual', 'requested' )` => exception
* `( 'requested', 'text', 'requested', 'source' )` => exception
* `( 'requested', 'text', 'requested', null )` => exception
* `( 'requested', 'text', 'requested', 'requested' )` => exception